caracal/workers/prep_worker.py

```diff
@@ -2,29 +2,110 @@
 import os
 import sys
 import caracal
+from caracal.dispatch_crew.utils import closeby
+import caracal.dispatch_crew.caltables as mkct
 import numpy as np
 from caracal.workers.utils import manage_flagsets as manflags
 from caracal.dispatch_crew import utils
+from astropy.coordinates import SkyCoord
 
 NAME = "Prepare Data for Processing"
 LABEL = 'prep'
 
+def getfield_coords(info, field, db, tol=2.9E-3, tol_diff=4.8481E-6):
+    """
+    Shameless copy of the hetfield function to return field coordinates 
+    from the database.
+    Find match of fields in info
+    Parameters:
+    info (dict): dictionary of obsinfo as read by yaml
+    field (str): field name
+    db (dict):   calibrator data base as returned by
+                 calibrator_database()
+    Go through all calibrators in db and return the first that matches
+    the coordinates of field in msinfo. Return empty string if not
+    found.
+    If coordinates difference is larger than tol_diff, return the correct coordinates, else return empty string.
+    """
+
+    # Get position of field in msinfo
+    ind = info['FIELD']['NAME'].index(field)
+    firade = info['FIELD']['DELAY_DIR'][ind][0]
+    firade[0] = np.mod(firade[0],2*np.pi)
+    dbcp = db.db
+    for key in dbcp.keys():
+        carade = [dbcp[key]['ra'],dbcp[key]['decl']]
+        if closeby(carade, firade, tol=tol):
+            print("Closeby source found")
+            if not closeby(carade, firade, tol=tol_diff):
+               return key, dbcp[key]['ra'], dbcp[key]['decl']
+            else :
+               return key, '', ''
+    return False
```

Comment:
``return None, None, None`` rather. Otherwise you complicate the error checking and get into unnecessary contortions (see comment below).

Reply:
On it.

Reply:
Fixed.